crates/networking/discv5/src/subnet.rs

```diff
@@ -1,173 +1,456 @@
+        Ok(())
+    }
+
+    pub fn disable_sync_committee_subnet(&mut self, subnet_id: u8) -> anyhow::Result<()> {
+        ensure!(
+            subnet_id < SYNC_COMMITTEE_SUBNET_COUNT as u8,
+            "Subnet ID {} exceeds maximum sync committee subnet count {}",
+            subnet_id,
+            SYNC_COMMITTEE_SUBNET_COUNT
+        );
+        self.set(subnet_id as usize, false)?;
+        Ok(())
+    }
+
+    pub fn is_sync_committee_subnet_enabled(&self, subnet_id: u8) -> anyhow::Result<bool> {
+        ensure!(
+            subnet_id < SYNC_COMMITTEE_SUBNET_COUNT as u8,
+            "Subnet ID {} exceeds maximum sync committee subnet count {}",
+            subnet_id,
+            SYNC_COMMITTEE_SUBNET_COUNT
+        );
+        self.get(subnet_id as usize)
     }
 }
 
-impl Encodable for Subnets {
+impl From<BitVector<U4>> for SyncCommitteeSubnets {
+    fn from(bits: BitVector<U4>) -> Self {
+        Self(bits)
+    }
+}
+
+impl Encodable for SyncCommitteeSubnets {
     fn encode(&self, out: &mut dyn BufMut) {
-        let ssz_bytes = self.as_ssz_bytes();
-        let bytes = Bytes::from(ssz_bytes);
+        let bytes = Bytes::from(self.0.as_ssz_bytes());
         bytes.encode(out);
     }
 }
 
-impl Decodable for Subnets {
+impl Decodable for SyncCommitteeSubnets {
     fn decode(buf: &mut &[u8]) -> alloy_rlp::Result<Self> {
         let bytes = Bytes::decode(buf)?;
-        let subnets = Subnets::from_ssz_bytes(&bytes).map_err(|err| {
+        let subnets = BitVector::<U4>::from_bytes(bytes.to_vec().into()).map_err(|err| {
             alloy_rlp::Error::Custom(Box::leak(
-                format!("Failed to decode SSZ subnets: {err:?}").into_boxed_str(),
+                format!("Failed to decode SSZ SyncCommitteeSubnets: {err:?}").into_boxed_str(),
             ))
         })?;
-        Ok(subnets)
+        Ok(Self(subnets))
     }
 }
 
-pub fn subnet_predicate(subnets: Vec<Subnet>) -> impl Fn(&Enr) -> bool + Send + Sync {
+pub fn attestation_subnet_predicate(subnets: Vec<u8>) -> impl Fn(&Enr) -> bool + Send + Sync {
     move |enr: &Enr| {
-        let Some(subnets_state) = (match enr
-            .get_decodable::<Subnets>(ATTESTATION_BITFIELD_ENR_KEY)
-            .transpose()
-        {
-            Ok(subnets_state) => subnets_state,
-            Err(err) => {
-                error!("Could not get subnets_state {err:?}, {}", enr.to_base64());
+        if subnets.is_empty() {
+            return true;
+        }
+
+        let attestation_bits =
+            match enr.get_decodable::<AttestationSubnets>(ATTESTATION_BITFIELD_ENR_KEY) {
+                Some(Ok(subnets)) => subnets,
+                _ => {
+                    trace!(
+                        "Peer rejected: invalid or missing attnets field; peer_id: {}",
+                        enr.node_id()
+                    );
+                    return false;
+                }
+            };
+
+        for subnet_id in &subnets {
```

Comment:
```suggestion
        for subnet_id in subnets {
```
the reason it was erroring earlier was because we can't take a reference to a slice